### PR TITLE
GDCPS-1709: Adding more logic to PresentationFrontStorageBus and next events

### PR DESCRIPTION
**Pull request by rkotangoor**
Updated presentationFrontStorageBus to go to transformFrontToBackfill as the next event , added front transform to get the backfillDoc key, add solr-adapter, added unit tests

### CODE REVIEW

File: lib/handlers/presentationFrontStorageBus.js [hunk 1]
```diff
@@ -12,11 +12,9 @@ module.exports = function presentationFrontStorageBus(next, state) {
         if (state.event.type === 'front-published') {
             state.transformedDocuments = [];
             state.upsertedDocuments = [];
-            state.aggregateType = 'front';
+            state.aggregateType = 'front-backfill';
```
Comment (scottgunther) [inline lib/handlers/presentationFrontStorageBus.js]: it should still be a 'front'. This is only to with what type of data: asset, staff, or front

File: lib/transforms/front.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+'use strict';
```
Comment (scottgunther) [inline lib/transforms/front.js]: please always use presentation-transforms. You can add this to that code base.

Comment (rkotangoor) [inline lib/transforms/front.js]: done, Made code changes in presentation-transforms, and updated to the latest version in here.

File: lib/handlers/presentationFrontStorageBus.js (hunk 1 above)
Comment (rkotangoor) [inline lib/handlers/presentationFrontStorageBus.js]: done. renamed state.aggregateType from `front-backfill` to `front`

File: lib/adapters/solr-adapter.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+var config = require('config');
+var _ = require('lodash');
+
+module.exports = new SolrCommunicator();
+
+function SolrCommunicator() {
+    this._clients = {};
+}
+
+SolrCommunicator.prototype._getClient = function(collection) {
```
Comment (scottgunther) [inline lib/adapters/solr-adapter.js]: I'd prefer to just use the solr client in presentation-index-handler. It is super simple and that is all that is needed. We don't have multiple collections we are searching.

Comment (rkotangoor) [inline lib/adapters/solr-adapter.js]: ok, makes sense, I will update.

Comment (rkotangoor) [inline lib/adapters/solr-adapter.js]: done.

File: sources.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "presentationFrontStorageBus": {
     "handler": "presentationFrontStorageBus.js",
+    "assetSearchSolr": "transformFrontToBackfill",
```
Comment (scottgunther) [inline sources.json]: I would try to make these names similar or do `success: transformFrontToBackfill`. Then change it in the next function call as well.

Comment (rkotangoor) [inline sources.json]: done

File: test/unit/lib/handlers/transformFrontToBackfill-tests.js [hunk 5]
```diff
@@ -0,0 +1,543 @@
...
+        });
+
+        it('should call presentationFrontTransform.getKeyBackfill with the passed in state.event.data backfillUtils.getPresentationFrontBackfillDocument returns layoutModuleDocs', function() {
+            backfillUtils = {
+                getPresentationFrontBackfillDocument: sinon.stub().returns({
+                    layoutModuleDocs: {123: [], 456: []},
+                    recommendedDate: recommendedDate
+                })
+            };
+            mockery.resetCache();
```
Comment (scottgunther) [inline test/unit/lib/handlers/transformFrontToBackfill-tests.js]: You should be able to modify the `backfillUtils.getPresentationFrontBackfillDocument` without needing to reset cache. Just modify it before the call to `transform`. If you are changing the `backfillUtils` itself then it is an issue but a property on the object should work.

Comment (scottgunther) [inline test/unit/lib/handlers/transformFrontToBackfill-tests.js]: Something like: `backfillUtils.getPresentationFrontBackfillDocument = sinon.stub().returns(...)`

Comment (rkotangoor) [inline test/unit/lib/handlers/transformFrontToBackfill-tests.js]: done and done.

**Pull request merged**